<?xml version="1.0"?>
<launch>


    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>

    <!-- Gazebo sim -->
    <!-- Needed for quad -->
    <include file="$(find quadnodes)/launch/gazeboOutdoor.launch">
      <arg name="world" default="$(find quadnodes)/worlds/empty.world"/>
      <arg name="gui" default="false"/>
    </include>


    <!-- Spawn UAV1 -->
    <include file="$(find quadnodes)/launch/spawnUAV1_SDF.launch">
      <arg name="x" default="$(arg x)"/>
      <arg name="y" default="$(arg y)"/>
      <arg name="z" default="$(arg z)"/>
    </include>

    <include file="$(find quadnodes)/launch/50x50.launch"/>

    <group ns="UAV1">
      <!-- Gas sensor -->
      <node pkg="simulated_gas_sensor" type="simulated_gas_sensor" name="fake_pid" output="screen">
          <param name="sensor_model" value="30" />
          <param name="sensor_frame" value="pid_frame1" />
          <param name="fixed_frame" value="map_gaden"/>
          <param name="noise_std" value="20.1"/>
      </node>

      <!-- Motion planner -->
      <node name="BRWQuad" pkg="quadnodes" type="BRWQuad.py" output="screen">
        <param name="minLim"                 value="0.0"   />   # m
        <param name="maxLim"                 value="50.0"  />   # m
        <param name="zHeight"                value="2.0"   />   # m
        <param name="biasRange"              value="60.0"  />   # degrees
        <param name="stepSize"               value="1.0"   />   # m
        <param name="waypointRadius"         value="0.25"  />   # m
        <param name="stayTime"               value="0.25"   />   # seconds
        <param name="maxVelocity"            value="3.0"   />   # seconds
        <param name="GaussPlume" type="bool" value="False" />   # I'm not explaining this
        <param name="GadenPlume" type="bool" value="True"  />   # I'm not explaining this
      </node>
    </group>

    <!-- Create tf transformations -->
    <node pkg="tf" type="static_transform_publisher" name="gaden2World" args="0.0 0.0 0.0 0 0 0 world map_gaden 100" />
    <node pkg="tf" type="static_transform_publisher" name="UAVs2World" args="0.0 0.0 0.0 0 0 0 world worldUAVs 100" />

    <node pkg="tf" type="static_transform_publisher" name="pid1_broadcaster" args="0.0 0.0 0.01 0 0 0 UAV1 pid_frame1 100" />

    <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot2"  args="/UAV1/Sensor_reading/raw"/> -->

    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find quadnodes)/rvizConfigs/gadenSingleQuad.rviz"/> -->

    <!-- <node pkg="rosbag" type="record" args="-o /home/matthew/google-drive/Gaden/gadenBRWSims/gadenBRWSim /UAV1/Sensor_reading /UAV1/true_position" name="rosbagRecord" /> -->

</launch>
